datafusion/functions/src/string/uuid.rs

```diff
@@ -87,9 +88,25 @@ impl ScalarUDFImpl for UuidFunc {
         if !args.is_empty() {
             return internal_err!("{} function does not accept arguments", self.name());
         }
-        let values = std::iter::repeat_with(|| Uuid::new_v4().to_string()).take(num_rows);
```

Comment:
This old code will allocate a new string, create uuid, then copy that string into the output. 

Removing that allocation alone is likely to make a big difference